test/index.js

```diff
@@ -270,6 +271,32 @@ function wrap(before) {
   };
 }
 
+const makeValues = () => ({
+  Array: ['?', '!'],
+  Boolean: new Boolean (true),
+  Date: new Date (0),
+  Function: x => x + 1,
+  Number: new Number (42),
+  Object: {'?': '!'},
+  RegExp: /foo/,
+  String: new String ('?!'),
+});
+
+const domesticValues = makeValues ();
+const alienValues = {};
+
+new vm.Script (`Object.assign(alienValues, (${makeValues.toString ()})())`)
+.runInNewContext ({alienValues});
```

Comment:
How about the following?

```suggestion
const alienValues = vm.runInNewContext (String (makeValues)) ();
```